### PR TITLE
Dimensionful quantities with Unitful.jl

### PR DESCRIPTION
**Pull request by carstenbauer**
See #7.

I left the custom unit output as I think `J/mol K` is nicer than `J mol^-1 K^-1`.

One could also make this an upstream issue in Unitful.jl and improve the printing of units (https://github.com/ajkeller34/Unitful.jl/issues/18 is similar).

### CODE REVIEW

**Comment by rahulkp220** [comment]
@crstnbr We can also add some test cases for this one.

**Comment by carstenbauer** [comment]
What kind of test do you have in mind? The unit printing is implicitly tested by the `stringmime` tests.

**Comment by carstenbauer** [comment]
Ok, I guess I can write some very basic ones.

**Comment by rahulkp220** [comment]
@crstnbr Yup, just the basic will work. Wanted to make sure we do have these tested out :)

**Comment by carstenbauer** [comment]
I'll merge and tag a new v0.1.2 release for this if you approve the changes.

**Review by rahulkp220 — APPROVED** [review]
LGTM. Are you good to have a new release post this merge? @crstnbr

**Comment by carstenbauer** [comment]
Maybe we should explicitly mention that the fields are dimensionful in the Readme.md as well. Will add this and proceed afterwards.

**Comment by rahulkp220** [comment]
Sure, makes sense. You can also reply to this comment [here](https://discourse.julialang.org/t/ann-new-release-of-periodictable-jl/14726/2?u=rahulkp220) once we are good with the release.

File: src/PeriodicTable.jl [hunk 1]
```diff
@@ -12,22 +12,23 @@ module PeriodicTable
...
     category::String
     color::String
-    density::Float64
+    density::typeof(1.0g/cm^3)
     discovered_by::String
     el_config::String
-    melt::Float64
-    molar_heat::Float64
+    melt::typeof(1.0K)
+    molar_heat::typeof(1.0J/(mol*K))
```
Comment (mschauer) [inline src/PeriodicTable.jl:31]: this reads nice, did not know this works

**Pull request merged**